sky/core.py

```diff
@@ -111,6 +111,32 @@ def status(cluster_names: Optional[Union[str, List[str]]] = None,
                                       cluster_names=cluster_names)
 
 
+def status_kubernetes():
+    """Get all SkyPilot clusters and controllers in the Kubernetes cluster.
+
+    Managed jobs and services are also included in the clusters returned.
+    The caller must parse the controllers to identify which clusters are run
+    as managed jobs or services.
+
+    Returns:
+        A tuple containing:
+        - List of dictionaries with cluster information.
+        - Dictionary of job controller information.
+        - Dictionary of serve controller information.
```

Comment:
Why do we use `List` for the first return, but `Dict` for the following? Can we keep them consistent, e.g., using `List` for all of them?

Also, using `dict` for the information is a bit fragile. I would vote to gradually change everything to `dataclasses.dataclass` or even `pydantic.BaseModel`. Could we use dataclass for the returned value of this (and the underlying function)?

Reply:
Good point - updated to return List in all cases. Also updated to use dataclass, was previously trying to be consistent with the return type from `backend_utils.get_clusters` :)